core/src/HIP/Kokkos_HIP_Abort.hpp

```diff
@@ -53,12 +53,13 @@
 namespace Kokkos {
 namespace Impl {
 
-__device__ inline void hip_abort(char const *msg) {
-  printf("%s", msg);
-  // FIXME_HIP both abort and the __assertfail system call are currently
-  // implemented with __builtin_trap which causes the program to exit abnormally
-  // without printing the error message.
-  // abort();
+__device__ __attribute__((noinline)) void hip_abort(char const *msg) {
+#ifndef NDEBUG
+  // disable printf on release builds, as it has a non-trivial performance
+  // impact
```

Comment:
Can you elaborate on how this would affect performance outside that function?

Reply:
Will discuss this morning